# king country house sales prediction:
   
 

BUSINESS OBJECTIVE:      
                    A real estate company in king country(seattle) wants to know about future sales prices  of houses for upcoming  housesales.
  
ANALYSIS:  
  
 I took previous years king country house sales dataset from kaggle.Performed datacleaning,exploratory data analysis and feature enginneering to extract the  important features from dataset and performed data visualization to present the reports of which attributes highly contribute to the prediction of  house prices. After analysis,the  machine learning model is generated to predict the sales prices using various regression analysis and their  performance is compared. 


1)importing packages and data collection:

In [10]:
### importing necessary packages
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import norm

In [11]:
## reading kc/-house_data
df=pd.read_csv('kc_house_data.csv')
print(type(df))
df.shape

<class 'pandas.core.frame.DataFrame'>


(21613, 21)

as seen above the dataset contains 21613 rows .(ie) entries and 21 columns .(ie) features

In [12]:
## lets look at  a sample of dataset
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

The above code list out first five rows of dataset.Its seen that price column(dependent feature) is the one we are trying to predict and remaining columns are independent features.lets look  deeper into dataset

2)data manipulation:

In [14]:
###  gets info about each feature
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
id               21613 non-null int64
date             21613 non-null object
price            21613 non-null float64
bedrooms         21613 non-null int64
bathrooms        21613 non-null float64
sqft_living      21613 non-null int64
sqft_lot         21613 non-null int64
floors           21613 non-null float64
waterfront       21613 non-null int64
view             21613 non-null int64
condition        21613 non-null int64
grade            21613 non-null int64
sqft_above       21613 non-null int64
sqft_basement    21613 non-null int64
yr_built         21613 non-null int64
yr_renovated     21613 non-null int64
zipcode          21613 non-null int64
lat              21613 non-null float64
long             21613 non-null float64
sqft_living15    21613 non-null int64
sqft_lot15       21613 non-null int64
dtypes: float64(5), int64(15), object(1)
memory usage: 3.5+ MB


Its seen from o/p, there is no missing values in datasets... and also we can remove unrelated features such as id,date,zipcode 

In [25]:
df.drop('id',axis=1,inplace=True)
df

KeyError: "['id'] not found in axis"

In [26]:
## lets see statistical analysis for each features
df.describe()

price      bedrooms     bathrooms   sqft_living      sqft_lot  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  2.161300e+04   
mean   5.400881e+05      3.370842      2.114757   2079.899736  1.510697e+04   
std    3.671272e+05      0.930062      0.770163    918.440897  4.142051e+04   
min    7.500000e+04      0.000000      0.000000    290.000000  5.200000e+02   
25%    3.219500e+05      3.000000      1.750000   1427.000000  5.040000e+03   
50%    4.500000e+05      3.000000      2.250000   1910.000000  7.618000e+03   
75%    6.450000e+05      4.000000      2.500000   2550.000000  1.068800e+04   
max    7.700000e+06     33.000000      8.000000  13540.000000  1.651359e+06   

             floors    waterfront          view     condition         grade  \
count  21613.000000  21613.000000  21613.000000  21613.000000  21613.000000   
mean       1.494309      0.007542      0.234303      3.409430      7.656873   
std        0.539989      0.086517      0.766318      0.650743      1.175459   
min        1.000000      0.000000      0.000000      1.000000      1.000000   
25%        1.000000      0.000000      0.000000      3.000000      7.000000   
50%        1.500000      0.000000      0.000000      3.000000      7.000000   
75%        2.000000      0.000000      0.000000      4.000000      8.000000   
max        3.500000      1.000000      4.000000      5.000000     13.000000   

         sqft_above  sqft_basement      yr_built  yr_renovated           lat  \
count  21613.000000   21613.000000  21613.000000  21613.000000  21613.000000   
mean    1788.390691     291.509045   1971.005136     84.402258     47.560053   
std      828.090978     442.575043     29.373411    401.679240      0.138564   
min      290.000000       0.000000   1900.000000      0.000000     47.155900   
25%     1190.000000       0.000000   1951.000000      0.000000     47.471000   
50%     1560.000000       0.000000   1975.000000      0.000000     47.571800   
75%     2210.000000     560.000000   1997.000000      0.000000     47.678000   
max     9410.000000    4820.000000   2015.000000   2015.000000     47.777600   

               long  sqft_living15     sqft_lot15  
count  21613.000000   21613.000000   21613.000000  
mean    -122.213896    1986.552492   12768.455652  
std        0.140828     685.391304   27304.179631  
min     -122.519000     399.000000     651.000000  
25%     -122.328000    1490.000000    5100.000000  
50%     -122.230000    1840.000000    7620.000000  
75%     -122.125000    2360.000000   10083.000000  
max     -121.315000    6210.000000  871200.000000

describe command describes each features count,central tendency,quartile ranges.For better understanding we analyse each feature statistical part in detail in visualization.

3) data exploration & visualization:
     
     EDA(exploratory data analysis) include ,
               1.univariate analysis
               2.bi variate and multivariate analysis
               3.missing values and outlier treatement
               4.hypothesis testing
               5.feature engineering
               
               
     In order to perform analysis ,we  list out categorical and numeric variables from dataset.
     
        price ,a numeric variable is dependent variable. Below are independent variables,
     
           1) categorical variables:bedrooms,bathrooms,condition,floors,grade
           2) numeric  variables:sqft_living,sqft_lot,sqft_above,sqft_basement,lat,long,sqft_living15,sqft_lot_15